[DK-282] 채팅 조회 API 구현

midasWorld opened the pull request:
## ✅  작업 단위

### [[DK-282] feat: 채팅 조회 API 구현](https://github.com/prgrms-web-devcourse/Team_04_SFam_BE/commit/29490526d23a95c00847945894e02fcd84fabc16)
- 신청자 혹은 작성자가 채팅 기록들 조회하는 기능 구현

### [[DK-282] test: 채팅 조회 API 테스트 코드 작성](https://github.com/prgrms-web-devcourse/Team_04_SFam_BE/commit/b817ab6969b4c85c52f9e22704dff1d4cc9edb7f)
- 신청 Giver, 채팅 서비스 테스트 코드 작성

[DK-282]: https://insta-kkyu.atlassian.net/browse/DK-282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DK-282]: https://insta-kkyu.atlassian.net/browse/DK-282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

midasWorld committed:
[DK-282] feat: 채팅 조회 API 구현

midasWorld committed:
[DK-282] test: 채팅 조회 API 테스트 코드 작성

HYEBPARK reviewed (approved):
동운님 수고하셨습니당 !!

HYEBPARK commented on src/main/java/com/kdt/team04/domain/matches/proposal/repository/MatchProposalRepository.java:
```diff
@@ -12,4 +13,7 @@ public interface MatchProposalRepository extends JpaRepository<MatchProposal, Lo
 	List<MatchProposal> findAllByMatchId(@Param("matchId") Long matchId);
+
+	@Query("SELECT mp FROM MatchProposal mp JOIN FETCH User u ON mp.user.id = u.id WHERE mp.id = :id")
+	Optional<MatchProposal> findMatchProposalById(@Param("id") Long id);
```
이 부분 네이밍 조금 헷갈렸는데 저만 그런거면 넘어가고 ... 
다른 분도 그러시면 이부분도 같이 생각해보면 좋을 것 같습니당 !

midasWorld replied:
`findProposalWithUserById` 로 변경하겠습니다!

pjh612 commented on src/main/java/com/kdt/team04/domain/matches/proposal/dto/MatchChatResponse.java:
```diff
@@ -1,9 +1,25 @@
+	) { }
+
+	public record Chats(
+		MatchProposalResponse.ChatMatch match,
```
MatchChattings 또는 Chattings
Chatting 등 명사형이 좋을 것 같아요

midasWorld replied:
`Chatting` 로 변경하겠슴둥

pjh612 commented on src/main/java/com/kdt/team04/domain/matches/proposal/service/MatchProposalGiverService.java:
```diff
@@ -27,4 +36,30 @@ public MatchProposalQueryDto findSimpleProposalById(Long id) {
+		MatchResponse.MatchAuthorResponse match
+			= matchService.findMatchAuthorById(matchProposal.getMatch().getId());
+
+		if (!Objects.equals(matchProposal.getUser().getId(), userId)
```
이렇게 복합적인 조건문이 들어갈 때에는 조건문을 함수로 빼서 어떤 조건인지 함수명으로써 표현하면 더욱 좋은 코드가 될 것 같습니다.

midasWorld replied:
굳.. 수정하겠습니다! 👍

pjh612 reviewed (changes requested):
몇가지 의견 드렸습니다!

midasWorld committed:
[DK-282] refactor: 코드 리뷰 피드백 반영
- Chat DTO 클래스명 명사로 변경 → Chatting
- Fetch join 잘못 사용한 부분 수정
- 메서드명 모호한 부분 수정 : findMatchProposalById() → findProposalWithUserById()
- 복합한 조건문 별도 메서드로 분리

pjh612 reviewed (approved):
고생하셨습니다..!

midasWorld committed:
[DK-282] merge with develop
